src/dynamics/vertical_coordinates.jl

```diff
@@ -5,21 +5,27 @@ export SigmaCoordinates
 """Sigma coordinates for the vertical coordinates, defined by their half layers.
 Sigma coordinates are currently hardcoded in Geometry.
 $(TYPEDFIELDS)."""
-@kwdef struct SigmaCoordinates{NF, VectorType} <: AbstractVerticalCoordinate
-    nlayers::Int = DEFAULT_NLAYERS
-    σ_half::VectorType = default_sigma_coordinates(nlayers)
+struct SigmaCoordinates{NF, VectorType} <: AbstractVerticalCoordinate
+    nlayers::Int
+    σ_half::VectorType
 
     SigmaCoordinates{T, V}(nlayers::Integer, σ_half::AbstractVector) where {T, V} = sigma_okay(nlayers, σ_half) ?
     new{T, V}(nlayers, σ_half) : error("σ_half = $σ_half cannot be used for $nlayers-level SigmaCoordinates")
 end
 
-# obtain nlayers from length of predefined σ_half layers
-SigmaCoordinates(σ_half::AbstractVector) = SigmaCoordinates{eltype(σ_half), typeof(σ_half)}(nlayers=length(σ_half)-1; σ_half)
-SigmaCoordinates(σ_half::AbstractRange) = SigmaCoordinates(collect(σ_half))
-SigmaCoordinates(nlayers::Integer = DEFAULT_NLAYERS) = SigmaCoordinates{Float64, Vector{Float64}}(; nlayers)
+# constructor using default sigma coordinates if only nlayers provided, also collect to allow for AbstractRange
+SigmaCoordinates(nlayers::Integer, σ_half::AbstractVector = default_sigma_coordinates(nlayers)) =
```

Comment:
isn't this one overwriting the internal constructor in l 12?

Reply:
There's no warning coming though, so I am wrong apparently

Reply:
No because the original constructor has no type constraints I believe. It's only that `@kwdef` defines keyword constructors that include the empty constructor `()` that should not be defined again.